source/windows/bcrypt_rsa.c

```diff
@@ -0,0 +1,411 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/cal/private/rsa.h>
+
+#include <aws/cal/cal.h>
+#include <aws/cal/private/der.h>
+#include <aws/common/encoding.h>
+
+#define WIN32_NO_STATUS
+#include <windows.h>
+#undef WIN32_NO_STATUS
+
+#include <bcrypt.h>
+#include <ntstatus.h>
+
+static BCRYPT_ALG_HANDLE s_rsa_alg = NULL;
+
+static aws_thread_once s_rsa_thread_once = AWS_THREAD_ONCE_STATIC_INIT;
+
+static void s_load_alg_handle(void *user_data) {
+    (void)user_data;
+    /* this function is incredibly slow, LET IT LEAK*/
+    NTSTATUS status = BCryptOpenAlgorithmProvider(&s_rsa_alg, BCRYPT_RSA_ALGORITHM, MS_PRIMITIVE_PROVIDER, 0);
```

Comment:
"LET IT LEAK" is worrying :) Maybe expand on what that means?

Reply:
this is actually the same comment we have in several other places, ex. https://github.com/awslabs/aws-c-cal/blob/main/source/windows/bcrypt_ecc.c#L29. 
We init the handle once per thread and then just let os clean up any associated resources when process exits, cause we really dont have a good way to tell once we are done using the handle

Reply:
Small correction, it’s once per process not once per thread. Thread_once is a misleading name but it’s what stdlibs tend to call this mechanism so I kept it that way.

Reply:
Oh good, per process is much more reasonable than per thread :) I'd still suggest not having a comment that says "LET IT LEAK" with no context. And how is the speed of the method relevant?

Reply:
yeah, my bad. that aws_thread_call_once always throws me off. It is once per process.

We want to avoid opening algo several times because of how expensive it is and we dont have a good indicator of when user is done with crypto. So instead we open it once and never close it, hence the "LEAK" part. I can add more context to the comment.

Reply:
Anyone that has an affinity for RAII is going to by default go and open an alg handle for construction and close it on destroy. We’ve learned from experience in c++ sdk, then in the crt, that that scales really badly. It’s basically a note to whomever comes after us either to refactor OR to port this code over to rust or something like that that that bad code was intentional and to not change it. I’m sure we could make it a less ominous message, and more descriptive though.

Reply:
That function does things like, allow AWS to install a kms client for certain crypto operations, that everything on the machine ends up using. It goes through com loaders and all sorts of stuff. It should only be called once per process.